File: notification-service/src/main/java/com/eirsteir/coffeewithme/notification/repository/NotificationRepository.java
package com.eirsteir.coffeewithme.notification.repository;

import com.eirsteir.coffeewithme.notification.domain.Notification;
import java.util.List;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;

public interface NotificationRepository extends JpaRepository<Notification, Long> {

  List<Notification> findAllByUser_idOrderByTimestamp(Long userId, Pageable pageable);
}

File: notification-service/src/main/kotlin/com/eirsteir/coffeewithme/notification/repository/NotificationRepository.kt
package com.eirsteir.coffeewithme.notification.repository

import com.eirsteir.coffeewithme.notification.domain.Notification
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository

interface NotificationRepository : JpaRepository<Notification, Long> {
    fun findAllByUser_idOrderByTimestamp(userId: Long, pageable: Pageable): List<Notification>
}